framework/common/encryption.cpp

```diff
@@ -223,33 +223,57 @@ bool hmac::digest(uint8_t *digest)
     return HMAC_Final(m_ctx, digest, &digest_length);
 }
 
-bool aes_encrypt(const uint8_t *key, const uint8_t *iv, uint8_t *data, uint32_t data_len)
+bool aes_encrypt(const uint8_t *key, const uint8_t *iv, uint8_t *plaintext, int plen,
+                 uint8_t *ciphertext, int &clen)
```

Comment:
We already discussed this - openssl takes `int`. And regarding the inout parameters - I normally would agree with you on that, but in this case I think it makes the code easier to understand since this is the way the internal openssl APIs work, with in-out clen.